### PR TITLE
Update the datasetNum label regardless of the type of the dataset

### PR DESCRIPTION
**Pull request by littletiger123**
Signed-off-by: littletiger123 <mg20330006@smail.nju.edu.cn>

### Ⅰ. Describe what this PR does
This PR lets the datasetNum calculates the number of the dataset regardless of the type of the dataset(such as alluxioruntime, jindoruntime)

### Ⅳ. Describe how to verify it
I test the code in different cases. Continue to create, continue to delete and create and delete at the same time for both two kinds of the dataset.

### Ⅴ. Special notes for reviews

### CODE REVIEW

File: pkg/ddc/alluxio/shutdown.go [hunk 1]
```diff
@@ -287,18 +286,12 @@ func (e *AlluxioEngine) destroyWorkers(expectedWorkers int32) (currentWorkers in
...
-					e.Log.Error(err, "The dataset number format error")
-					return err
-				}
-				if currentDataset < 2 {
-					delete(toUpdate.Labels, labelDatasetNum)
-					labelNames = append(labelNames, labelDatasetNum)
-				} else {
-					toUpdate.Labels[labelDatasetNum] = strconv.Itoa(currentDataset - 1)
-				}
+			isDeleted, err := lifecycle.DatasetNumUpdateOrDelete(toUpdate, runtimeInfo)
```
Comment (cheyang) [inline pkg/ddc/alluxio/shutdown.go:289]: DatasetNumUpdateOrDelete -> UpdateOrDeleteDatasetNum

File: pkg/utils/dataset/lifecycle/node.go [hunk 2]
```diff
@@ -258,3 +258,22 @@ func labelNodeWithCapacityInfo(toUpdate *v1.Node, runtimeInfo base.RuntimeInfoIn
 	toUpdate.Labels[totalCapacityLabel] = totalValue
 	*updatedLabels = append(*updatedLabels, totalCapacityLabel)
 }
+
+// UpdateOrDeleteDatasetNum deletes the datasetNum label or updates the number of the dataset in the specific node.
+func UpdateOrDeleteDatasetNum(toUpdate *v1.Node, runtimeInfo base.RuntimeInfoInterface) (isDeleted bool, err error) {
```
Comment (cheyang) [inline pkg/utils/dataset/lifecycle/node.go:263]: Could you add test for this function? No need to do it today.

Comment (littletiger123) [reply pkg/utils/dataset/lifecycle/node.go:263]: Okay, I will pay more attention to it later.

**Review by yangyuliufeng — APPROVED** [review]

File: pkg/utils/dataset/lifecycle/node.go (hunk 2 above)
Comment (cheyang) [inline pkg/utils/dataset/lifecycle/node.go:263]: I suggest the name of the function rename to `DecreaseDatasetNum`？ WTDY？

**Review by cheyang — CHANGES_REQUESTED** [review]

**Review by cheyang — APPROVED** [review]

**Pull request merged**